### PR TITLE
[MISC] Update release template

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -26,27 +28,37 @@ assignees: ''
...
 
-  4. Alter the file `/web/docs.seqan.de/htdocs/index.html` with a link to the new documentation build.
+  4. Alter the file `/web/docs.seqan.de/htdocs/seqan3.html` with a link to the new documentation build.
 
   </details>
-- [ ] Prepare `seqan-[VERSION]-with-submodules.tar.gz`.
+- [ ] Prepare `seqan3-[VERSION]-{Linux,Source}.tar.xz{,.sha265}`.
   <details><summary>Click for detailed steps</summary><br>
 
   Make a fresh clone of the repository and recursively delete the `.git` directories.
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:43]: What do we need to delete now? Only the main `.git` from the checkout?

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -26,27 +28,37 @@ assignees: ''
...
   ```
+  mkdir package-build
+
   git clone https://github.com/seqan/seqan3.git
   cd seqan3
+
+  git checkout 3.0.2 # version to pack
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:50]: Shouldn't this be the release branch?
The release tag does not exist before tagging...

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -57,21 +69,54 @@ assignees: ''
...
 
-- [ ] **Tag release on Github and attach `seqan-[VERSION]-with-submodules.tar.gz` to the release.**
+- [ ] **Tag release on Github and attach `seqan3-[VERSION]-{Linux,Source}.tar.xz{,.sha265}` to the release.**
+
+  <details><summary>Click for detailed steps</summary><br>
+
+  Github is not able to create annotated releases (https://github.com/seqan/product_backlog/issues/159), so we have to manually sign the release.
+  ```bash
+  git checkout release-3.x.y
+  git tag -s 3.x.y
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:79]:
```suggestion
  git tag -a -s 3.x.y
```
Does `-s` imply `-a`?
With what certificate is the tag signed?

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 4]
```diff
@@ -57,21 +69,54 @@ assignees: ''
...
+  git push upstream 3.x.y
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:80]:
```suggestion
  git push upstream 3.x.y
```
What about `git push --tags`?

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 5]
```diff
@@ -57,21 +69,54 @@ assignees: ''
...
+  You can find a comprehensive list of the changes in our [changelog](http://docs.seqan.de/seqan/3.0.2/about_changelog.html).
+  ```
+
+  </details>
 
 ---
 
-- [ ] Bump succeeding version number. ([include/seqan3/version.hpp](https://github.com/seqan/seqan3/blob/6391aaa014db4d04b8220a2041014295c4134f33/include/seqan3/version.hpp#L19-L24))
+- [ ] Merge release branch into master.
+- [ ] Bump succeeding version number in master. ([include/seqan3/version.hpp](https://github.com/seqan/seqan3/blob/6391aaa014db4d04b8220a2041014295c4134f33/include/seqan3/version.hpp#L19-L24))
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:105]: Double check link or at least update link to current release

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 6]
```diff
@@ -7,9 +7,11 @@ assignees: ''
 
 ---
 
-- [ ] Check for critical performance regressions.
-- [ ] Check [Nightlies](https://cdash.seqan.de/index.php?project=SeqAn3) for critical build failures.
-- [ ] Check the [Changelog.md](https://github.com/seqan/seqan3/blob/master/CHANGELOG.md) for completeness.
+- [ ] Create a release branch, e.g. `release-3.0.3`
```
[the remark this answers is not in the document]
Comment (eseiler) [reply .github/ISSUE_TEMPLATE/release-checklist.md:10]: We mix it throughout the release tasks... I'm slightly in favor of `[VERSION]`

[the remark this answers is not in the document]
Comment (marehr) [reply .github/ISSUE_TEMPLATE/release-checklist.md:10]: in favour for `[VERSION]`

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (marehr) [reply .github/ISSUE_TEMPLATE/release-checklist.md:43]: Nothing needs to be delete

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 2 above)
Comment (marehr) [reply .github/ISSUE_TEMPLATE/release-checklist.md:50]: yeah change it to `release-[VERSION]`

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 3 above)
Comment (marehr) [reply .github/ISSUE_TEMPLATE/release-checklist.md:79]: yes -s implies -a

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 4 above)
Comment (marehr) [reply .github/ISSUE_TEMPLATE/release-checklist.md:80]: that would push all tags, but you want to push only one tag (you might push other tags accidentally)

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (rrahn) [reply .github/ISSUE_TEMPLATE/release-checklist.md:10]: And please add to make it protected in the branch settings.

[the remark this answers is not in the document]
Comment (rrahn) [reply .github/ISSUE_TEMPLATE/release-checklist.md:10]: Is there anything to be done to make it visible in the nightlies?

[the remark this answers is not in the document]
Comment (eseiler) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: > Is there anything to be done to make it visible in the nightlies?

What do you mean?
To build the release branch, I need to change the source checkout from master to release branch, nothing else.

**Review by marehr — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 7]
```diff
@@ -26,52 +28,96 @@ assignees: ''
...
-- [ ] Bump succeeding version number. ([include/seqan3/version.hpp](https://github.com/seqan/seqan3/blob/6391aaa014db4d04b8220a2041014295c4134f33/include/seqan3/version.hpp#L19-L24))
+- [ ] Merge release branch into master.
+- [ ] Bump succeeding version number in master. ([include/seqan3/version.hpp](https://github.com/seqan/seqan3/blob/3.0.2/include/seqan3/version.hpp#L19-L24))
 - [ ] Update the SeqAn version [here](https://github.com/OpenMS/usage_plots/blob/master/seqan_versions.txt) to ensure
       that the server in Tübingen, which takes care of the argument parser update notifications, is aware of it
       (@smehringer).
-- [ ] Announce release on twitter.
-- [ ] Announce release on [www.seqan.de](https://www.seqan.de).
+- [ ] Delete release branch.
+- [ ] Announce release on [twitter](https://twitter.com/seqanlib).
```
Comment (eseiler) [inline .github/ISSUE_TEMPLATE/release-checklist.md:111]:
```suggestion
- [ ] Delete release branch.
- [ ] Change [Nightlies](http://cdash.seqan.de/index.php?project=SeqAn3) back to master branch.
- [ ] Announce release on [twitter](https://twitter.com/seqanlib).
```

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 8]
```diff
@@ -7,9 +7,11 @@ assignees: ''
 
 ---
 
-- [ ] Check for critical performance regressions.
-- [ ] Check [Nightlies](https://cdash.seqan.de/index.php?project=SeqAn3) for critical build failures.
-- [ ] Check the [Changelog.md](https://github.com/seqan/seqan3/blob/master/CHANGELOG.md) for completeness.
+- [ ] Create a release branch, e.g. `release-[VERSION]`, and enable branch protection in the reposity settings.
+- [ ] Check for critical performance regressions. Use https://github.com/wvandertoorn/cmp_benchmarks
```
Comment (marehr) [inline .github/ISSUE_TEMPLATE/release-checklist.md:11]:
```suggestion
- [ ] Create at least one pre-Release s.t. our package maintainers can build a new update of our library
- [ ] Check for critical performance regressions. Use https://github.com/wvandertoorn/cmp_benchmarks
```

**Review by rrahn — COMMENTED** [review]
some pending comments

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 9]
```diff
@@ -7,9 +7,12 @@ assignees: ''
 
 ---
 
-- [ ] Check for critical performance regressions.
-- [ ] Check [Nightlies](https://cdash.seqan.de/index.php?project=SeqAn3) for critical build failures.
-- [ ] Check the [Changelog.md](https://github.com/seqan/seqan3/blob/master/CHANGELOG.md) for completeness.
+- [ ] Create a release branch, e.g. `release-[VERSION]`, and enable branch protection in the reposity settings.
+- [ ] Create at least one pre-Release s.t. our package maintainers can try out a new release of our library (they will need a complete source-package to do this).
+- [ ] Check for critical performance regressions. Use https://github.com/wvandertoorn/cmp_benchmarks
```
Comment (rrahn) [inline .github/ISSUE_TEMPLATE/release-checklist.md:12]:
```suggestion
- [ ] Check that the directory structure is valid: https://github.com/seqan/seqan3/blob/master/doc/setup/quickstart_cmake/index.md
- [ ] Check for critical performance regressions. Use https://github.com/wvandertoorn/cmp_benchmarks
```

**Review by rrahn — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 10]
```diff
@@ -26,52 +29,96 @@ assignees: ''
...
 - [ ] Announce release on mailing list `seqan-dev@lists.fu-berlin.de`.
-- [ ] Announce release on public Gitter channel [![Gitter](https://badges.gitter.im/seqan/Lobby.svg)](https://gitter.im/seqan/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge).
+- [ ] Announce release on [public Gitter channel](https://gitter.im/seqan/Lobby).
 - [ ] Notify upstream package maintainers
-  - brew (@rrahn)
-  - macports (@rrahn)
-  - conda (@eseiler)
-  - debian (@mr-c)
+  - [ ] brew (@rrahn)
+  - [ ] macports (@rrahn)
```
Comment (rrahn) [inline .github/ISSUE_TEMPLATE/release-checklist.md:119]:
```suggestion
  - [ ] macports (@rrahn)
  - [ ] bio.tools (@rrahn)
```

**Comment by eseiler** [comment]
@rrahn I would propose to add links to the packages we maintain ourselves? 
That way we know where the brew, macports, bio.tools and conda acutally lives...

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 11]
```diff
@@ -26,52 +30,97 @@ assignees: ''
...
+- [ ] Announce release on [website](https://www.seqan.de).
 - [ ] Announce release on mailing list `seqan-dev@lists.fu-berlin.de`.
-- [ ] Announce release on public Gitter channel [![Gitter](https://badges.gitter.im/seqan/Lobby.svg)](https://gitter.im/seqan/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge).
+- [ ] Announce release on [public Gitter channel](https://gitter.im/seqan/Lobby).
 - [ ] Notify upstream package maintainers
-  - brew (@rrahn)
-  - macports (@rrahn)
-  - conda (@eseiler)
-  - debian (@mr-c)
+  - [ ] brew (@rrahn)
```
Comment (rrahn) [inline .github/ISSUE_TEMPLATE/release-checklist.md:119]:
```suggestion
  - [ ] brew [https://github.com/brewsci/homebrew-bio/tree/develop/Formula](@rrahn)
```

Comment (rrahn) [reply .github/ISSUE_TEMPLATE/release-checklist.md:119]: maybe rather as markdown link.

**Review by rrahn — APPROVED** [review]

**Pull request merged**